conans/client/rest/rest_client_common.py

```diff
@@ -242,7 +242,10 @@ def search(self, pattern=None, ignorecase=True):
         """
         url = self.router.search(pattern, ignorecase)
         response = self.get_json(url)["results"]
-        return [ConanFileReference.loads(reference) for reference in response]
+        try:
+            return [ConanFileReference.loads(reference) for reference in response]
+        except TypeError as te:
+            raise Exception("Unexpected response from server url: `%s`." % url)
```

Comment:
It would probably be useful to display the ``response`` too, to give some idea of what could be happening.

Reply:
I agree. However, I felt that the TypeError raised alongside the custom Exception would show the response type. (It is my understanding that both Exceptions will be included in the Traceback. In this issue's case; None, or any other type that is not iterable in a list-comp. I felt that the response type, along with the URL, would be sufficient information to assist with a debug.
Happy to reconsider though... 👍

Reply:
> I agree. However, I believe that the TypeError raised alongside the custom Exception would show the response type. (It is my understanding that both Exceptions will be included in the Traceback.)

Many exceptions do not display a stack trace. As Conan is more an app than a library, we try to hide internal implementation details as much as possible, and most of the exceptions are captured into a ``ConanException`` that only displays the error message, but not the stack trace (I am not sure in this case if that Exception might be captured somewhere into a ConanException). The idea is that in most cases, if Conan is raising an exception that it knows what is happening, it should raise a ConanException. So it is always better try to provide an error message that is as user-oriented and self-contained as possible. In this case showing the response could help, because there are sometimes that the URL is incorrect, and return html, or there is a proxy returning something in the response too. IN those cases, printing that to the user might be useful information.

Reply:
Ah, I got it. User-friendly error messages, not dev friendly. Will refactor, and look into using a `ConanException` instead. Thanks.